cranelift/wasm/src/environ/spec.rs

```diff
@@ -648,6 +648,52 @@ pub trait FuncEnvironment: TargetEnvironment {
     fn use_x86_pmaddubsw_for_dot(&self) -> bool {
         false
     }
+
+    /// Inserts call to check_malloc and check_free before the return of malloc
+    /// and free if wasmtime is compiled and run with wmemcheck.
```

Comment:
Let's write these from the generic perspective of Cranelift, without knowledge of wasmtime or wmemcheck (here we're defining the interface/tools that the higher abstraction layer uses). So something like "Insert code before a function return."

Likewise below: before loads/stores/global updates/memory growths.